src/common/collection.c

```diff
@@ -2480,7 +2480,7 @@ static void _dt_collection_recount_callback_1(gpointer instance, gpointer user_d
   {
     if(old_count != collection->count) dt_collection_hint_message(collection);
     DT_DEBUG_CONTROL_SIGNAL_RAISE(darktable.signals, DT_SIGNAL_COLLECTION_CHANGED, DT_COLLECTION_CHANGE_RELOAD,
-                                  DT_COLLECTION_PROP_UNDEF, NULL, -1);
+                                  DT_COLLECTION_PROP_UNDEF, (void *)0, -1);
```

Comment:
Is that really necessary? I mean we want to pass a pointer to a list here (GList *), so when not defined it seems clearer to use NULL which is what we use to initialize a list usually:

`   GList *list = NULL;`

So at the end, this is obfuscating the code to me.

Reply:
This comes from

> CWE: 475
Passing NULL after the last typed argument to a variadic function leads to undefined behaviour.
The C99 standard, in section 7.15.1.1, states that if the type used by va_arg() is not compatible with the type of the actual next argument (as promoted according to the default argument promotions), the behavior is undefined.
The value of the NULL macro is an implementation-defined null pointer constant (7.17), which can be any integer constant expression with the value 0, or such an expression casted to (void*) (6.3.2.3). This includes values like 0, 0L, or even 0LL.

There a few more places in the codebase where this is found, might add them here before merging?

Reply:
> (GList *)NULL would also work if you want to avoid obfuscation.

Yes please, this looks better to me.